packages/core/src/stateUtils.ts

```diff
@@ -15,12 +15,16 @@ export const isLeafNode = (
   stateNode: StateNode<any, any, any, any, any, any>
 ) => stateNode.type === 'atomic' || stateNode.type === 'final';
 
+export function getAllChildren<TC, TE extends EventObject>(
```

Comment:
we probably could name those 2 in a better way - but the overall algorithm is overhauled in v5 so I didn't bother too much with figuring out better names right now